rcl/src/rcl/client.c

```diff
@@ -316,13 +312,9 @@ rcl_take_response(
 bool
 rcl_client_is_valid(const rcl_client_t * client)
 {
-  const rcl_client_options_t * options;
-  RCL_CHECK_ARGUMENT_FOR_NULL(client, false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(client, "client pointer is invalid", return false);
   RCL_CHECK_FOR_NULL_WITH_MSG(
     client->impl, "client's rmw implementation is invalid", return false);
-  options = _client_get_options(client);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    options, "client's options pointer is invalid", return false);
```

Comment:
What's the rationale for removing the check of the options? Does it prevent this functions reuse somewhere else or something?

Reply:
I thought it was redundant since `options` is not a pointer in the underlying impl struct, which is already checked in the line above.
The private function `_client_get_options()` returns the address, and the user-facing `rcl_client_get_options()` calls `rcl_client_is_valid()` anyways.

Reply:
Makes sense, 👍